menas/src/main/scala/za/co/absa/enceladus/menas/services/VersionedModelService.scala

```diff
@@ -72,6 +72,66 @@ abstract class VersionedModelService[C <: VersionedModel with Product with Audit
     versionedMongoRepository.getLatestVersionValue(name)
   }
 
+  def exportSingleItem(name: String, version: Int): Future[String] = {
+    getVersion(name, version).flatMap({
+      case Some(item) => Future(item.exportItem())
+      case _ => throw NotFoundException()
+    })
+  }
+
+  def exportLatestItem(name: String): Future[String] = {
+    getLatestVersion(name).flatMap({
+      case Some(item) => Future(item.exportItem())
+      case _ => throw NotFoundException()
+    })
+  }
+
+  def importSingleItem(item: C, username: String, metadata: Map[String, String]): Future[Option[C]] = {
+    for {
+      validation <- validateSingleImport(item, metadata)
+      result <- {
+        if (validation.isValid()) {
+          importItem(item, username)
+        } else {
+          throw ValidationException(validation)
+        }
+      }
+    } yield result
+  }
+
+  private[services] def validateSingleImport(item: C, metadata: Map[String, String]): Future[Validation] = {
+    val validation = Validation()
+      .withErrorIf(!hasValidNameChars(item.name), "item.name", s"name '${item.name}' contains unsupported characters")
+      .withErrorIf(item.parent.isDefined, "item.parent", "parent should not be defined on import")
+    val withMetadataValidation = validateMetadata(validation, metadata)
+    Future(withMetadataValidation)
+  }
+
+  private[services] def validateMetadata(validation: Validation, metadata: Map[String, String]): Validation = {
+    val exportVersionErrorMessage =
```

Comment:
I would make this an inner function used as a parameter into `withErrorIf`.
`def exportVersionErrorMessage(version: String).` Then perhaps the ugly `.stripMargin` won't be needed even.

Reply:
Fixed